src/yellow/components/button/_index.scss

```diff
@@ -130,24 +165,45 @@
       border-top: 3px solid transparent;
       box-sizing: border-box;
       content: '';
-      height: 20px;
+      height: 18px;
       left: 50%;
+      margin: -9px 0 0 -9px;
       position: absolute;
       top: 50%;
-      transform-origin: 0 0;
-      width: 20px;
+      width: 18px;
+
+      @keyframes spin {
+        100% {
+          transform: rotate(360deg);
+        }
+      }
     }
-  }
 
-  &--full-card {
-    border-bottom-left-radius: $size-border-radius-m;
-    border-bottom-right-radius: $size-border-radius-m;
-    border-top: get-border(size-border-width-none);
-    border-top-left-radius: 0;
-    border-top-right-radius: 0;
-  }
+    &[class~='btn--inverse'],
+    &[class~='btn--secondary'] {
+      &::before {
+        border-bottom: 3px solid $color-interface-content-dark;
+        border-left: 3px solid $color-interface-content-dark;
+        border-right: 3px solid $color-interface-content-dark;
+      }
+    }
+
+    &[class~='btn--inverse'][class~='btn--secondary'] {
+      &::before {
+        border-bottom: 3px solid $color-interface-content-light;
+        border-left: 3px solid $color-interface-content-light;
+        border-right: 3px solid $color-interface-content-light;
+      }
 
-  &--rounded {
-    border-radius: 30px;
+      &:hover,
+      &:focus,
+      &:active {
```

Comment:
Lines 182 and 183 can be merged with these rules because they do the same thing

Reply:
Not really @TrevelinT, on line 182 and 183 I see if the button is `btn - inverse` **OR**` btn - secondary`, here I am checking if `btn - secondary` **also contains**` btn - inverse` .

Reply:
My mistake. I didn't pay attention to the indentation. The `hover`, `focus`  and `active` is part of the `&[class~='btn--inverse'][class~='btn--secondary']` class